### ISSUE TITLE
segmentation fault in connect with multiple threads

### ISSUE THREAD

**Issue by jakobj** [nest/nest-simulator#211]
using multiple threads, when calling connect with a mixture of neuron and devices, a segmentation fault occurs as threads create connections in non-thread-local sparse tables.
the following lines will generate a segmentation fault:

``` postscript
0 << /total_num_virtual_procs 2 /print_time false >> SetStatus

/n1 /iaf_psc_delta Create def
/n2 /iaf_psc_delta Create def
/n3 /iaf_psc_delta Create def

/sd /spike_detector Create def

[n1 n1] [sd n3] Connect
```

in this case thread0 and thread1 both try to create a connection in the connector of neuron 1 on thread 1. this can be traced back to the use of  `get_node(gid, thread)` in the connection builders (`AllToAllBuilder` etc.) which defaults to thread=0 if no argument is given.
most likely this bug was not ecountered before since users rarely mix devices and neurons in a single list. the segmentation fault occurs with 2.10.0, 2.8.0, 2.6.0 and i did not go back further in time.

am i right in assuming this is not the desired behaviour?

ps: i am now working on a fix based on the nest3 branch.

**Comment by jakobj** [comment]
there still seem to be old connection routines like DivergentConnect in the connection_builder_manager. i thought these were completely replaced by the new connect framework? does it make sense to fix them aswell as they exist for a good reason or should i ignore them? @heplesser @hannahbos

**Comment by jougs** [comment]
We decided to do the restructuring first and then clean up for the release once everything is in master.

**Comment by heplesser** [comment]
This has been fixed by #221, thus closing.
